### ISSUE TITLE
SteamClient.SteamId is terribly slow

### ISSUE THREAD

**Issue by kristianpiacenti** [Facepunch/Facepunch.Steamworks#784]
It's not cached and do weird things behind the scenes. It should be moved to a Get method not to cause confusion.
Calling this property each frame can destroy your cpu time.

It's used also in Friend.IsMe so be careful for that.

**Comment by DK189** [comment]
For `SteamAPI_ISteamUser_GetSteamID`, the delay is probably because SteamAPI actually calls this data response from SteamClient every time it's requested. I don't understand why they would do that either, for security? It's obviously not supposed to change while the game is running.
Account name, level, avatar, and other things need to be refreshed every time it's called. `SteamID` doesn't need it.

**Comment by kristianpiacenti** [comment]
You are right @DK189 , it's not suppose to change unless you log-in with a different account, but you lose authentication if you try something like and the client should shut down automatically.

I propose just cache that in SteamClient, something simple like

```
private static bool _steamIdIsCached;
private static SteamId _steamIdCache;
public static SteamId SteamId
{
	get
	{
		if (_steamIdIsCached == false)
		{
			_steamIdIsCached = true;
			_steamIdCache = SteamUser.Internal.GetSteamID();
		}
		
		return _steamIdCache;
	}	
}
```

and keep a _RequestSteamId()_ in case you need to be sure you have the right one.
(and at that point for consistency you have RequestSteamId in SteamId property)

**Comment by DK189** [comment]
Yes! All programmers should see this! 👍👍👍
